src/components/tools/geometry-tool/geometry-toolbar.tsx

```diff
@@ -0,0 +1,124 @@
+@inject("stores")
+@observer
+export class GeometryToolbarView extends BaseComponent<IProps, IState> {
+  constructor(props: IProps) {
+    super(props);
+    this.state = {
+      showSettings: false,
+    };
+  }
+  public render() {
+    return (
+      <div className="geometry-toolbar" data-test="geometry-toolbar" onMouseDown={this.handleMouseDown}>
+        <div className="toolbar-buttons">
+          {this.renderToolHeader(false)}
+          {this.renderToolButton("Select", "selection", null,
+                                  false, false, true)}
+          {this.renderToolButton("Point", "point", null,
+                                  false, false, true)}
+          {this.renderToolButton("Polygon", "polygon", null,
+                                  false, false, true)}
+          {this.renderToolButton("Duplicate", "duplicate", this.props.onDuplicateClick,
+                                  false, this.props.duplicateDisabled, false)}
+          {this.renderToolButton("Line Label", "line-label", null,
+                                  false, false, true)}
+          {this.renderToolButton("Angle Label", "angle-label", this.props.onAngleLabelClick,
+                                  this.props.angleLabelSelected, this.props.angleLabelDisabled, false)}
+          {this.renderToolButton("Movable Line", "movable-line", null,
+                                  false, false, true)}
+          {this.renderToolButton("Draw", "draw", null,
+                                  false, false, true)}
+          {this.renderToolButton("Text", "text", null,
+                                  false, false, true)}
+          {this.renderToolButton("Delete", "delete", this.props.onDeleteClick,
+                                  false, this.props.deleteDisabled, false)}
+        </div>
+        { this.state.showSettings
+          ? <div className="settings">
+              <div className="settings-header">
+                Settings
+              </div>
+            </div>
+          : null
+        }
+      </div>
+    );
+  }
+
+  private renderToolHeader = (showSettings: boolean) => {
+    return (
+      <div className="toolbar-header">
+        <div className="graph-tile">
+          <svg className="header-icon">
+            <use xlinkHref="#icon-graph-tile"/>
+          </svg>
+        </div>
+        {showSettings
+          ? <div className="settings-button" title="Settings"
+                 onClick={this.handleSettingsButton}>
+              <svg className="icon">
+                <use xlinkHref="#icon-settings"/>
+              </svg>
+            </div>
+          : null
+        }
+      </div>
+    );
+  }
+
+  private renderToolButton = (toolName: string, toolClass: string, handleClick: any,
+                              selected: boolean, disabled: boolean, hidden: boolean) => {
+    let buttonClass = "button " + toolClass + " ";
+    buttonClass += selected ? "selected " : "";
+    buttonClass += disabled ? "disabled " : "enabled ";
```

Comment:
I'd be inclined to use [classnames](https://www.npmjs.com/package/classnames) to reduce this to:
```typescript
const buttonClass = classNames("button", toolClass, { selected, disabled, enabled: !disabled });
```

Reply:
Did you try:
```typescript
import * as classNames from "classnames";
```
That seems to be the way to import modules without a default export in TypeScript, a la `import * as React from "react";`.

Reply:
That along with adding `declare module 'classnames';` to `typing.d.ts` does the trick.

Reply:
Did you try `npm install @types/classnames` instead of `declare module 'classnames';`? Given how popular the module is I'd be surprised if no one's added TypeScript types for it.

Reply:
Reverts back to a Typescript error without the declare: `Cannot find module 'classnames'.`